src/main/java/com/google/sps/servlets/utils/Statistics.java

```diff
@@ -22,62 +22,102 @@
 import java.util.stream.Collectors;
 
 public class Statistics {
-  private static final double LOWER_END_VAL = -1.0;
-  private static final double UPPER_END_VAL = 1.0;
-  private static final BigDecimal INTERVAL = BigDecimal.valueOf(0.2);
-  private static final BigDecimal UPPER_END = BigDecimal.valueOf(UPPER_END_VAL);
-  private static final BigDecimal LOWER_END = BigDecimal.valueOf(LOWER_END_VAL);
+  private static final double LOWER_SCORE_VAL = -1.0;
+  private static final double UPPER_SCORE_VAL = 1.0;
+  private static final double LOWER_MAGNITUDE_VAL = 0.0;
+  private static final double UPPER_MAGNITUDE_VAL = 2.0;
+  private static final double SCORE_INTERVAL_VAL = 0.2;
+  private static final double MAGNITUDE_INTERVAL_VAL = 0.3;
+  private static final BigDecimal SCORE_INTERVAL = BigDecimal.valueOf(SCORE_INTERVAL_VAL);
+  private static final BigDecimal UPPER_SCORE = BigDecimal.valueOf(UPPER_SCORE_VAL);
+  private static final BigDecimal LOWER_SCORE = BigDecimal.valueOf(LOWER_SCORE_VAL);
+  private static final BigDecimal MAGNITUDE_INTERVAL = BigDecimal.valueOf(MAGNITUDE_INTERVAL_VAL);
+  private static final BigDecimal LOWER_MAGNITUDE = BigDecimal.valueOf(LOWER_MAGNITUDE_VAL);
+  private static final BigDecimal UPPER_MAGNITUDE = BigDecimal.valueOf(UPPER_MAGNITUDE_VAL);
 
-  // Contains sentiment scores in the range [-1, 1] with given intervals.
-  private Map<Range, Integer> aggregateValues;
+  // Contains sentiment scores and magnitudes with given intervals.
+  private Map<Range, Integer>
```

Comment:
I think we should determine a new format for how data should be represented within this class.

Since this class is directly serialized to JSON, we'll want to define a format that is easily serializable.

I've talked with Molly a little bit about this, but one disadvantage of serializing Maps with JSON is that JSON maps only support `String` keys so having Range as a key makes deserializing more difficult (as it forces us to parse the String representation of the `Range` instead of being able to consume it as JSON.)

Since the front end logic is currently relying upon the `aggregateValues` having a specific format, we shouldn't necessarily change that yet, but we should likely aim to have some kind of `List` organization instead of `Map`.

The exact structure of the data that we should pass to the front should be driven by what we want to display there, so I'm hesitant to make any specific suggestions until it's more clear what extensions you guys want to make.